src/wren_core.c

```diff
@@ -878,6 +878,24 @@ DEF_PRIMITIVE(num_cos)
   RETURN_NUM(cos(AS_NUM(args[0])));
 }
 
+DEF_PRIMITIVE(num_decimal)
+{
+  double value = AS_NUM(args[0]);
+  RETURN_NUM(value - (int32_t)value);
```

Comment:
This will do the wrong thing on double values that don't fit in an `int32_t`. I'd use `modf()` from math.h.